src/main/java/Response/ResponseFormatter.java

```diff
@@ -0,0 +1,27 @@
+public class ResponseFormatter {
+  Response response;
+  
+  public ResponseFormatter(Response response) {
+    this.response = response;
+  }
+  
+  public String formatResponse() {
+    return formatStatusLine() + 
+           formatHeaders() + 
+           "\r\n" +
+           this.response.getMessageBody();
+  }
+
+  private String formatStatusLine() {
+    return formatHttpVersion() + " " + this.response.getStatusCode() + " " + this.response.getReasonPhrase() + "\r\n";
+  }
+
+  private String formatHttpVersion() {
+    return "HTTP/" + this.response.getHTTPVersion();
+  }
+
+  private String formatHeaders() {
+    return ResponseHeader.CONTENT_TYPE + ": " + this.response.getContentType() + "\n" +
+           ResponseHeader.CONTENT_LENGTH + ": " + this.response.getContentLength()+ "\r\n";
+  }
```

Comment:
I think this can be a lot more generic - if you use a map for headers, you can iterate over the map to create the string.

Reply:
I like this idea, but I'm concerned about how I'll test this class. As mentioned above, the Handlers in the `ft/image-content-negotiation` class populate `messageBody` with `byte[]`. The public method in that class (`#formatResponse()`) also returns a `byte[]`, so I there I have to `assertTrue` that the expected response bytes and the actual response bytes are equal using `Arrays.equals()`. Thus, the formatted response will differ depending on the order in which the method iterates over the `headers` HashMap. For example, the way that my code is written out now, `FileHandler` outputs `Content-Length` before `Content-Type`. Originally, I asserted equality with the expectation that `Content-Type` would appear before `Content-Length`. I see three ways around this. 

1. Check for the inclusion of certain bytes in the `byte[]` array returned by `formatResponse()`. For example, my test can look something like: 
```Java
@Test 
public void formatsResponseWithHeaders() {
  ResponseFormatter formatter = new ResponseFormatter(response);        
  byte[] expectedHeaderBytes =  "Content-Type: text/plain".getBytes();
  byte[] responseBytes = formatter.formatResponse();
  assertTrue(Arrays.asList(responseBytes).contains(expectedHeaderBytes));
}
```
I don't think that it's a very clear test, and it ultimately fails to test that the entire response is formatted correctly. I can also convert the entire response to a String and assert against an expected String, but the issue about the order remains. 

2. Move `#formatHeader()` to the `Response` class, exposing it publicly and testing it. This still seems a little beyond the class's responsibility because it has to know about where to include a CRLF, and it doesn't solve the issue about the order. 

3. Keep `#formatHeaders()` in `ResponseFormatter`, but only set one header to ensure consistency in the order of the headers.

Reply:
This should not make testing difficult (assuming you're talking about the ResponseFormatter tests). In your tests, you will build a response to format. Your test knows the order in which the headers are added, so it knows what to expect.

And none of your other tests should be looking at the response bytes, so this won't affect them either.